Shield.Communication/Services/Wifi.cs

```diff
@@ -0,0 +1,297 @@
+                }
+                catch (Exception e)
+                {
+                    // ignore dropped listen sockets
+                }
+            }
+        }
+
+        private async void ListenerOnMessageReceived(DatagramSocket sender, DatagramSocketMessageReceivedEventArgs args)
+        {
+            object outObj;
+            if (CoreApplication.Properties.TryGetValue("remotePeer", out outObj))
+            {
+                EchoMessage((RemotePeer)outObj, args);
+                return;
+            }
+
+            // We do not have an output stream yet so create one.
+            try
+            {
+                IOutputStream outputStream = await listener.GetOutputStreamAsync(args.RemoteAddress, args.RemotePort);
+
+                // It might happen that the OnMessage was invoked more than once before the GetOutputStreamAsync completed.
+                // In this case we will end up with multiple streams - make sure we have just one of it.
+                RemotePeer peer;
+                lock (this)
+                {
+                    if (CoreApplication.Properties.TryGetValue("remotePeer", out outObj))
+                    {
+                        peer = (RemotePeer)outObj;
+                    }
+                    else
+                    {
+                        peer = new RemotePeer(outputStream, args.RemoteAddress, args.RemotePort);
+                        CoreApplication.Properties.Add("remotePeer", peer);
+                    }
+                }
+
+                EchoMessage(peer, args);
+            }
+            catch (Exception exception)
+            {
+                // If this is an unknown status it means that the error is fatal and retry will likely fail.
+                if (SocketError.GetStatus(exception.HResult) == SocketErrorStatus.Unknown)
+                {
+                    throw;
+                }
+
+               // NotifyUserFromAsyncThread("Connect failed with error: " + exception.Message, NotifyType.ErrorMessage);
+            }
+
+        }
+
+        async void EchoMessage(RemotePeer peer, DatagramSocketMessageReceivedEventArgs eventArguments)
+        {
+            if (!peer.IsMatching(eventArguments.RemoteAddress, eventArguments.RemotePort))
+            {
+                // In the sample we are communicating with just one peer. To communicate with multiple peers application
+                // should cache output streams (i.e. by using a hash map), because creating an output stream for each
+                //  received datagram is costly. Keep in mind though, that every cache requires logic to remove old
+                // or unused elements; otherwise cache turns into a memory leaking structure.
+                //NotifyUserFromAsyncThread(String.Format("Got datagram from {0}:{1}, but already 'connected' to {3}", eventArguments.RemoteAddress, eventArguments.RemotePort, peer), NotifyType.ErrorMessage);
```

Comment:
Remove commented out code (or comment why it is commented out, and under what circumstances to add it back)